frontend/src/Settings.js

```diff
@@ -1,9 +1,105 @@
-import React from "react";
+import React, { useState } from 'react';
+import { Grid } from '@material-ui/core';
+import * as constants from './common/constants';
+import Snackbar from '@material-ui/core/Snackbar';
+import Alert from '@material-ui/lab/Alert';
+import axios from 'axios';
+import ProcessorCard from './common/components/ProcessorCard';
+
+const initState = () => {
+  /*In future releases, it should be a fetched array from the API*/
+  return {
+    processors: [
+      {
+        host: localStorage.getItem(constants.processorHostStorageKey)
+          || '0.0.0.0',
+        port: localStorage.getItem(constants.processorPortStorageKey)
+          || '8300'
+      }
+    ]
+  };
+};
 
 export default function Settings() {
-    return (
-        <div>
-            Settings
-        </div>
-    );
+  const [state, setState] = useState(initState());
+
+  const saveConfig = async (processor, index) => {
+    if (!processor) {
+      setState({
+        ...state,
+        showMsg: true,
+        severity: 'error',
+        alertMsg: 'All fields are required'
+      });
+      return;
+    }
+
+    try {
+      const response = await axios.put('/config', {
```

Comment:
Here you are making a request to the processor API? In that case, You should use the processor host and port as host for the request. And the params you have to send here are the host and port of the camera the node should process.

Like:
PUT http://{processor_host}:{processor_port}/config?video_path={video_url}

(Maybe not url encoded but to get the idea)

Reply:
Got it now, changing it